lib/ddtrace/contrib/rails/framework.rb

```diff
@@ -43,17 +47,12 @@ def self.setup
             activate_action_view!(datadog_config, rails_config)
             activate_active_record!(datadog_config, rails_config)
           end
-
-          # Update the tracer if its not the default tracer.
-          if rails_config[:tracer] != Datadog.configuration.tracer
-            rails_config[:tracer].default_service = rails_config[:service_name]
-          end
         end
 
-        def self.config_with_defaults
+        def self.config_with_defaults(datadog_config)
           # We set defaults here instead of in the patcher because we need to wait
           # for the Rails application to be fully initialized.
-          Datadog.configuration[:rails].tap do |config|
+          datadog_config[:rails].tap do |config|
```

Comment:
This seems weird... functionally speaking, it think your change is basically identical. However, it seems to me that there's a collision where `#[](key)` resolves an integration where `#[](key)` might also be used to resolve a configuration option too. I guess we only use direct function calls on configuration options (e.g. `configuration.tracer.hostname`)?

Probably no change necessary, just pointing this out.